R/format-public-data.R

```diff
@@ -0,0 +1,171 @@
+#'Format public data
+#'
+#'Format the merged trips into files that are fit for distribution.
+#'Specifically, this function produces a trips and a catch table as well as
+#'tables containing aggregated data. These files are stored both as tsv files
+#'and rds in a storage bucket with public access.
+#'
+#'#'The parameters needed in the config file are those required for
+#'`merge_trips()` in addition to:
+#'
+#'```
+#'public_storage:
+#'  google:
+#'    key: gcs
+#'    options:
+#'      project:
+#'      bucket:
+#'      service_account_key:
+#'export:
+#'  file_prefix:
+#'```
+#'@param log_threshold
+#' @inheritParams ingest_landings
+#'@keywords workflow
+#'@return no outputs. This function is used for it's side effects
+#'@export
+#'
+format_public_data <- function(log_threshold = logger::DEBUG){
+
+  logger::log_threshold(log_threshold)
+  pars <- read_config()
+
+  logger::log_info("Retrieving merged trips...")
+  merged_trips <- get_merged_trips(pars)
+
+  logger::log_info("Calculating summary fields")
+  merged_trips_with_addons <- add_calculated_fields(merged_trips)
+
+  logger::log_info("Creating trips table")
+  trips_table <- get_trips_table(merged_trips_with_addons)
+  logger::log_info("Creating catch table")
+  catch_table <- get_catch_table(merged_trips_with_addons)
+
+  logger::log_info("Aggregating data")
+  aggregated <- c("day", "week", "month", "year") %>%
+    rlang::set_names() %>%
+    purrr::map(summarise_trips, merged_trips_with_addons)
```

Comment:
That's a very nice output here, but it's not clear to me how it works! hope you can briefly explain me these lines